server/database.py

```diff
@@ -392,7 +394,22 @@ def recent_bans(self, count=5):
                         ORDER BY ban_date DESC LIMIT ?)
                     ORDER BY ban_date ASC
                     '''), (count,)).fetchall()]
-
+    def log_buffer(self,message):
+        if self.buffer_counter == 500:
+            self.buffer_counter = 0
+        else:
+            self.buffer_counter = self.buffer_counter + 1
+        self.message_buffer[self.buffer_counter] = message
```

Comment:
I do not understand the purpose of this circular buffer. Just query the database for the last 500 messages.